src/tiles/tile-group.component.ts

```diff
@@ -61,18 +66,32 @@ export class TileGroup implements AfterContentInit {
 	onTouched = () => { };
 
 	ngAfterContentInit() {
-		this.selectionTiles.forEach(tile => {
-			tile.name = this.name;
-			tile.change.subscribe(() => {
-				this.selected.emit({
-					value: tile.value,
-					selected: tile.selected,
-					name: this.name
-				});
-				this.onChange(tile.value);
+		const updateTiles = () => {
+			this.selectionTiles.forEach(tile => {
+				tile.name = this.name;
+				tile.change
+					.pipe(takeUntil(this.unsubscribe$))
+					.subscribe(() => {
+						this.selected.emit({
+							value: tile.value,
+							selected: tile.selected,
+							name: this.name
+						});
+						this.onChange(tile.value);
+					});
+				tile.multiple = this.multiple;
 			});
-			tile.multiple = this.multiple;
-		});
+		};
+		updateTiles();
+
+		this.selectionTiles.changes
+			.pipe(takeUntil(this.unsubscribe$))
+			.subscribe(_ => updateTiles());
```

Comment:
You'll probably want to do a `this.unsubscribe$.next` here ... otherwise you may run into multiple `selected` events emitting

Reply:
Wouldn't that unsubscribe right away? It happens onDestroy currently.

Reply:
I mean, currently nothing is every _actually_ unsubscribed, it just stops processing events ... I'm actually not sure why `this.selectionTiles.changes` has a `takeUntil` ... it should be cleaned up/completed by angular in the destroy phase and any subscriptions will also stop receiving when the final `unsubscribe` is emitted.